pkger/clone_resource.go

```diff
@@ -31,6 +37,300 @@ func (r ResourceToClone) OK() error {
 	return nil
 }
 
+var kindPriorities = map[Kind]int{
+	KindLabel:                         1,
+	KindBucket:                        2,
+	KindCheck:                         3,
+	KindCheckDeadman:                  4,
+	KindCheckThreshold:                5,
+	KindNotificationEndpoint:          6,
+	KindNotificationEndpointHTTP:      7,
+	KindNotificationEndpointPagerDuty: 8,
+	KindNotificationEndpointSlack:     9,
+	KindNotificationRule:              10,
+	KindTask:                          11,
+	KindVariable:                      12,
+	KindDashboard:                     13,
+	KindTelegraf:                      14,
+}
+
+type exportKey struct {
+	orgID influxdb.ID
+	name  string
+	kind  Kind
+}
+
+func newExportKey(orgID influxdb.ID, k Kind, name string) exportKey {
+	return exportKey{
+		orgID: orgID,
+		name:  name,
+		kind:  k,
+	}
+}
+
+type exportCoordinator struct {
+	bucketSVC   influxdb.BucketService
+	checkSVC    influxdb.CheckService
+	dashSVC     influxdb.DashboardService
+	labelSVC    influxdb.LabelService
+	endpointSVC influxdb.NotificationEndpointService
+	ruleSVC     influxdb.NotificationRuleStore
+	taskSVC     influxdb.TaskService
+	teleSVC     influxdb.TelegrafConfigStore
+	varSVC      influxdb.VariableService
+
+	mObjects map[exportKey]Object
+}
+
+func (ex *exportCoordinator) Export(ctx context.Context, resourcesToClone []ResourceToClone, labelNames ...string) error {
+	cloneAssFn, err := ex.resourceCloneAssociationsGen(ctx, labelNames...)
+	if err != nil {
+		return err
+	}
+
+	resourcesToClone = uniqResourcesToClone(resourcesToClone)
+	sort.Slice(resourcesToClone, func(i, j int) bool {
+		iName, jName := resourcesToClone[i].Name, resourcesToClone[j].Name
```

Comment:
this order here, provides guarantees that dependencies for a resource (associations) are accounted for. I'll add a comment to make that obvious.